### DISCUSSION TITLE
Integration w/ Brokerage API

### DISCUSSION THREAD

**Discussion by ethanj20** [skfolio/skfolio#45] [Ideas]
Is an integration with a brokerage API (for example, Schwab's [Trader API](https://developer.schwab.com/products/trader-api--individual) on the roadmap? Would be interesting to be able to:
1. Generate an optimized portfolio
2. Make appropriate trades to optimize the portfolio in my brokerage

Right now it seems the flow would be reviewing the results from my skfolio analysis, then manually submitting trades in my brokerage account to match the optimized portfolio, correct?

Thanks in advance, and apologies for any naivety as I am just getting up to speed.

**Comment by HugoDelatte** [comment]
Your understanding is correct. An integration with a brokerage API is not in the roadmap and is also out-of-scope for this library.
`skfolio` focuses on model selection and validation and the shipping to production is a task left to the user (same as `scikit-learn`). The main reason is that they are too many ways to ship a model to production.
However, nothing stop us or someone else to create a satellite package that ships `skfolio` models to production for a specific broker.
